src/sinks/file/mod.rs

```diff
@@ -88,98 +95,143 @@ impl PartitionedFileSink {
 
         Some(bytes)
     }
-}
-
-impl Sink for PartitionedFileSink {
-    type SinkItem = Event;
-    type SinkError = ();
-
-    fn start_send(&mut self, event: Self::SinkItem) -> StartSend<Self::SinkItem, Self::SinkError> {
-        if let Some(key) = self.partition_event(&event) {
-            let encoding = self.encoding.clone();
 
-            let partition = self
-                .partitions
-                .entry(key.clone())
-                .or_insert_with(|| File::new(key.clone(), encoding.clone()));
-
-            self.last_accessed.insert(key.clone(), Instant::now());
+    fn deadline_at(&self) -> Instant {
+        Instant::now()
+            .checked_add(self.idle_timeout)
+            .expect("unable to compute next deadline")
+    }
 
-            partition
-                .start_send(event)
-                .map_err(|error| error!(message = "Error writing to partition.", %error))
-        } else {
-            Ok(AsyncSink::Ready)
+    async fn run(&mut self, input: impl Stream<Item = Event> + Send + Sync) -> crate::Result<()> {
+        pin_mut!(input);
+        loop {
+            tokio02::select! {
+                event = input.next() => {
```

Comment:
we could prob drop the extra brackets surrounding this.

Reply:
Yep. I wonder why doesn't autoformatter kick in. I guess it's another drawback of using macros.
I'll correct it.

Reply:
rustfmt can't format macro_rules unfortunately

Reply:
This branch got removed with a whole wrapping layer, so this is now obsolete.